### DISCUSSION TITLE
Unable to resolve client names from the local IP addresses

### DISCUSSION THREAD

**Discussion by meminens** [AdguardTeam/AdGuardHome#7152] [Q&A]
Hey everyone,

I’m fairly new to networking and DNS, so bear with me if my question seems a bit basic!

I've got AdGuard Home set up, and while the ad-blocking works perfectly, I’m running into an issue: it only shows local IP addresses instead of client names. Everything else is working flawlessly.

Here's a rundown of my setup, which might help pinpoint what I'm missing:

- **Router**: I’m using a Deco (TP-Link) M5 mesh system as my DHCP server. It's running the default firmware, and it's up-to-date.
- **AdGuard Home**: It’s connected to the router via Ethernet and has a static IP address (192.168.1.120). I haven’t set AdGuard Home as the DHCP server, but I did add the DHCP server's IP (192.168.1.1) in both the Upstream DNS Servers and Private Reverse DNS servers settings. I also enabled these checkboxes:

  ![Screenshot](https://github.com/user-attachments/assets/db6d8578-d121-4156-8246-5e495b9c72ea)

- **Client Names**: When I check the Deco app, I can see client names just fine, so the router can resolve hostnames.
- **DNS Lookup Issue**: Running `nslookup <device ip>` gives me this error:

  ```
  nslookup 192.168.1.118

  Server:		192.168.1.120
  Address:	192.168.1.120#53

  ** server can't find 118.1.168.192.in-addr.arpa: NXDOMAIN
  ```

- I don't have anything configured in the DNS Rewrites setting.

- **Tailscale**: I'm also using Tailscale on the same server as AdGuard Home to block ads on the go, and it’s working well. I've set this server as a Subnet and entered its local IP in Tailscale’s admin settings, enabling the "Override local DNS" option. Disabling Tailscale doesn’t seem to change anything, though.

If you need any more details to help me out, please let me know. I’d really appreciate any insights or suggestions on what I might be doing wrong.

Thanks a lot in advance for your help!

**Comment by he-pennypacker** [comment]
This is an issue with the TP-Link Deco, not AGH. I face the same limitation with my X55 mesh. See here:

https://community.tp-link.com/en/home/forum/topic/246292

What I have done is set up IP address reservations in the Deco app, then add clients in AGH using those IP addresses as identifiers. In the query log I then get something like this as the client:

192.168.1.10
mypcname

Clunky workaround, but the blame is squarely with TP-Link on this one unfortunately.

**Comment by meminens** [comment]
Thanks for responding. At least you confirmed that it is a TP-Link issue. I wasn't quite sure. I doubt they will fix it though.
